### ISSUE TITLE
java.lang.UnsupportedOperationException: Implement OTraverseStatement.refersToParent()

### ISSUE THREAD

**Issue by rajgopalv** [orientechnologies/orientdb#8592]
## Expected behavior  
`select from V let $type = (traverse * from $parent.$current.children)` should work without any error

## Actual behavior  
```
orientdb {db=test}> select from V let $type = (traverse * from $parent.$current.children)
Error: com.orientechnologies.orient.enterprise.channel.binary.OResponseProcessingException: Exception during response processing
Error: java.lang.UnsupportedOperationException: Implement OTraverseStatement.refersToParent()

```
## Steps to reproduce  
Connect to console or Connect to Studio and try to execute this query
`select from V let $type = (traverse * from $parent.$current.children)`

**Comment by luigidellaquila** [comment]
Hi @rajgopalv 

Thank you for reporting, I'll check it ASAP and let you know

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @rajgopalv 

I just pushed a fix for that specific problem, it will be released with v 3.0.9

Please consider that `$parent.$current.children` is not allowed as a target for a query, you have to do something as follows instead:

```
select from V let $type = (traverse * from (SELECT expand($parent.$parent.$current.children))
```

Thanks

Luigi
